### DISCUSSION TITLE
[Docker-Install] :  [error] cannot initialize ipset: ipset: dialing netfilter: dialing v4: socket: invalid argument

### DISCUSSION THREAD

**Discussion by MilesTEG1** [AdguardTeam/AdGuardHome#3008] [Q&A]
Hello,
I run Adguard-Home for severals months now, installed in a docker container on my Synology NAS, without problems.
I saw recently on the container log an error :
```log
2021/04/25 19:35:52 [error] cannot initialize ipset: ipset: dialing netfilter: dialing v4: socket: invalid argument
```
It don't seems to cause any troubles, but I'm wondering why I get this error. And I would like to correct it.

I don't understand the only passage [in the documentation](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains) that tell about ipset :

- `ipset` — List of domain-ipset_name associations for adding IP addresses of the specified domain names to an ipset list. Syntax: `DOMAIN[,DOMAIN].../IPSET_NAME[,IPSET_NAME]`.... IPv4 addresses are added to an ipset list with `ipv4` family, IPv6 addresses - to `ipv6` ipset list. ipset list must exist. This setting is supported on Linux only. This feature is similar to `--ipset` in dnsmasq.

In my config fille `AdGuardHome.yaml`, I have this :
```yaml
  ipset: []
```

Thanks for any help you can provide 😃 
Bye

**Comment by ameshkov** [comment]
> It don't seems to cause any troubles, but I'm wondering why I get this error. And I would like to correct it.

That's kinda okay for a Docker container. @ainar-g maybe we shouldn't log this as error since this is expected behavior in many cases?

**Reply by ainar-g** [reply]
I think we should still log it, but on the `debug` level and with an additional message explaining that this is not a critical error.  It would still be useful for cases where users want to debug why `ipset` doesn't work for them, and I would rather not lose the information about the fact that an error has, in fact, occurred.

**Reply by ameshkov** [reply]
info level is also okay, it just should be clear that this is not an error.

**Reply by MilesTEG1** [reply]
Hello,
Thanks for your reassuring informations.
I can let this error be there without concern, hoping it will be renamed as Info as you said, in maybe the next update :)

**Reply by ainar-g** [reply]
Yes, it will be fixed in the next beta and in v0.106.0 final release.

**Reply by MilesTEG1** [reply]
Do you have an ETA for v0.106 final release ?
There is some things I'd like to test, but I don't want to switch to dev builds.

**Reply by ainar-g** [reply]
The ETA is today, heh.

**Reply by MilesTEG1** [reply]
I installed the new version :D Thanks.
And now, the message is an info warning :

```log
2021/04/28 20:34:47 [info] warning: cannot initialize ipset: ipset: dialing netfilter: dialing v4: socket: invalid argument
```

👍🏻 

And for my knowledge, what cause this "error" with a docker installation ? (I'm not a pro of docker 😅 )

**Reply by ainar-g** [reply]
We're not sure, to be honest, but we do have a TODO about it.
